kata_agent.go

```diff
@@ -335,6 +410,45 @@ func (k *kataAgent) startPod(pod Pod) error {
 		hostname = hostname[:maxHostnameLen]
 	}
 
+	//
+	// Setup network interfaces and routes
+	//
+	networkNS, err := pod.storage.fetchPodNetwork(pod.id)
+	if err != nil {
+		return err
+	}
+	interfaces, routes, err := k.generateKataInterfacesAndRoutes(networkNS)
+	if err != nil {
+		return err
+	}
+	for _, ifc := range interfaces {
+		// send update interface request
+		ifcReq := &grpc.UpdateInterfaceRequest{
+			Interface: ifc,
+		}
+		resultingInterface, err := k.sendReq(ifcReq)
+		if err != nil {
+			k.Logger().WithFields(logrus.Fields{
+				"interface requested": fmt.Sprintf("%+v", ifc),
+				"resulting interface": fmt.Sprintf("%+v", resultingInterface),
+			}).WithError(err).Error("update interface request failed")
+			continue
```

Comment:
Why don't we fail here ? Have we agreed on ignoring failures here ?

Reply:
I agree on logging the error and continuing to apply remaining interfaces and routes.  I think that this is an error, but not necessarily catastrophic -- user can get into the workload and then assess what failed in setup.  WDYT?

Reply:
Well, only logging without returning this error might be confusing since the user will expect everything to work as expected if he does not check the logs. And he might not understand why he does not get what he expects inside the container.
We could have a retry mechanism (maybe retry 5 times with `100ms` between each time), so that we really make sure there is a real issue here, that would worth throwing an error.

Reply:
@sameo @jodh-intel WDYT ?

Reply:
@egernst @sboeuf If an interface or a route failed to be set, it's likely that networking will be at least partly dysfunctional, leaving the end-user with a potentially non working pod. By only logging this and not failing we'd give a broken pod to the upper layers of the stack and pretend it's all fine, which is bad from a end user experience perspective. In other words, we should log **and** fail here and the pod must not be passed upward. It should be stopped at least.

Reply:
+1 for failing - if we continue, surely the system is in an unknown state (bad)? Turning it round, do we ever think this might fail? If so, why?

Reply:
Agreed. Thanks guys.

Reply:
I should have stopped with error in the first place, mea culpa.  It won't work today as I didn't handle conflicting routes (for the initial interface route in the guest).    I have a new PR, https://github.com/kata-containers/agent/pull/148, R on the agent that I started (not ready just yet) which will act on a set of routes instead, acting fully declarative.  Without this I don't think we can really test networking in a container without bypassing conflict errors (because the initial interface route already exists).  This'll simplify our handling quite a bit --   I'll try to get this ready for merge today.